src/controllers/oauth.js

```diff
@@ -11,19 +17,35 @@ const oauth = new OAuthServer({ model: authModel })
 const oauthController = module.exports = {}
 
 oauthController.getAuthorizePage = (req, res, next) => {
-  // get query params to find out what the app is (client_id is required)
   if (!req.query || !req.query.client_id) {
-    return res.boom.badRequest('no client_id provided')
+    return res.boom.badRequest(errMessages.NO_CLIENT_ID)
+  }
+  if (!req.query.redirect_uri) {
+    return res.boom.badRequest(errMessages.NO_REDIRECT_URI)
+  }
+  if (!req.query.state) {
```

Comment:
Okay, I have been hitting my head against this trying to work out if we need it.

One the one hand, it seems like a pretty key security measure (http://www.thread-safe.com/2014/05/the-correct-use-of-state-parameter-in.html), on the other, github does not set it to be a required field. Not really sure what to take from that..

I have not used this parameter when building a client with gh oauth in the past, I expect it adds another level of complexity for the implementer, if it is done properly.

I think we should prob configure auth2-server to allow empty state: https://github.com/oauthjs/node-oauth2-server/blob/master/lib/handlers/authorize-handler.js#L62, and drop the constraint here.

Thoughts?

Reply:
I think maybe state is only so important in the 'implicit flow' method of OAuth2, i.e. not the 'authorization code' method. So maybe we should remove it.

Reply:
but I don't think it adds much complexity at all, I think we should only remove it if it causes trouble for app developers.